File: semester-6/rsi/lab1/part2/task1/xmlrpcclient/src/UserInteraction.java
import java.util.InputMismatchException;
import java.util.List;
import java.util.Scanner;

public class UserInteraction {
    public static int selectOption (String label, List<String> options) {
        System.out.println(label + "\n");

        for (int i = 0; i < options.size(); i++) {
            System.out.println((i + 1) + ". " + options.get(i));
        }

        int selectedIndex = UserInteraction.getInt("", 1, options.size());

        return selectedIndex -1;
    }

    public static int getInt (String question, int lowerBound, int upperBound) {
        Integer result = null;
        Scanner scan = new Scanner(System.in);

        while (result == null || result < lowerBound || result > upperBound) {
            System.out.println(question);
            try {
                result = scan.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Invalid input");
                scan.nextLine();
            }
        }

        return result;
    }

    public static String getStr (String question) {
        Scanner scan = new Scanner(System.in);
        System.out.println(question);
        return scan.nextLine();
    }
}

File: bachelor/semester-6/rsi/lab2/part2/gRPCclient/UserInteraction.cs
using System;
using System.Collections.Generic;

namespace gRPCclient
{
    class UserInteraction
    {
        public static int SelectOption(string label, List<string> options)
        {
            Console.WriteLine(label + "\n");

            for (int i = 0; i < options.Count; i++)
            {
                Console.WriteLine($"{i + 1}. {options[i]}");
            }

            return GetInt("", 1, options.Count) - 1;
        }


        public static int GetInt(string question, int lowerBound, int upperBound)
        {
            int result;

            Console.WriteLine(question);
            while (!int.TryParse(Console.ReadLine(), out result) || result < lowerBound || result > upperBound)
            {
                Console.WriteLine(question);
            }

            return result;
        }

        public static string GetStr(string question)
        {
            Console.WriteLine(question);
            return Console.ReadLine();
        }
    }
}